### PR TITLE
Update codegen with azcore breaking changes

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Modeled error types are no longer used so the Error() method has been
removed along with the raw field.  In addition the *HandleError()
methods for API have also been removed.
LROs no longer require an error handler func during construction.
The arguments to NewPipeline() and NewResponseError() have slightly
changed.
The client endpoint field in the options struct was renamed to Endpoint.
Removal of error types is pending M4 fixes however the plubming is in
place.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Partially fixes https://github.com/Azure/autorest.go/issues/730.  The modeled errors are no longer emitted as error types or used, however M4 bugs prevent removal of them.

File: src/generator/operations.ts [hunk 1]
```diff
@@ -946,15 +942,15 @@ function generateResponseUnmarshaller(op: Operation, response: SchemaResponse, u
     // use the designated time type for unmarshalling
     unmarshallerText += `\tvar aux *${response.schema.language.go!.internalTimeType}\n`;
     unmarshallerText += `\tif err := runtime.UnmarshalAs${getMediaType(response.protocol)}(resp, &aux); err != nil {\n`;
-    unmarshallerText += `\t\treturn ${zeroValue}, runtime.NewResponseError(err, resp)\n`;
+    unmarshallerText += `\t\treturn ${zeroValue}, runtime.NewResponseError(resp)\n`;
```
Comment (jhendrixMSFT) [inline src/generator/operations.ts:945]: This is incorrect as the unmarshalling error is lost.

Comment (jhendrixMSFT) [reply src/generator/operations.ts]: This has been fixed.

File: src/transform/transform.ts [hunk 2]
```diff
@@ -518,65 +517,51 @@ function processOperationResponses(session: Session<CodeModel>) {
...
-        const schemaError = (<SchemaResponse>ex).schema;
-        if (isObjectSchema(schemaError)) {
-          for (const prop of values(schemaError.properties)) {
-            // adding the Inner prefix on error types, since errors in Go have an Error() method
-            // in order to implement the error interface. This causes errors to not be able to have
-            // an Error field as well, since it would cause confusion
-            if (prop.language.go!.name === 'Error') {
-              prop.language.go!.name = 'Inner' + prop.language.go!.name;
+      // TODO: make this work once relevant M4 bugs have been fixed
+      // annotate all exception types as errors so we can skip generating them
```
Comment (chlowell) [inline src/transform/transform.ts:521]: Do you want to link the issues here?

**Review by chlowell — APPROVED** [review]
Does M4 also prevent a half measure like unexporting the unwanted types?

File: src/transform/transform.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply src/transform/transform.ts:521]: I haven't dug into this fully yet and opened them as Tim was OOF at the time.  I'm tracking this overall work in https://github.com/Azure/autorest.go/issues/730

**Comment by jhendrixMSFT** [comment]
There's also https://github.com/Azure/autorest.go/issues/757 which talks about dropping unused types.

**Pull request merged**